diff --git a/src/shell_manager.cc b/src/shell_manager.cc
index 0bccbdf5a..4412d4cd5 100644
--- a/src/shell_manager.cc
+++ b/src/shell_manager.cc
@@ -78,7 +78,7 @@ ShellManager::ShellManager(ConstArrayView<EnvVarDesc> builtin_env_vars)
     // have access to the kak command regardless of if the user installed it
     {
         const char* path = getenv("PATH");
-        auto new_path = format("{}../libexec/kak:{}", split_path(get_kak_binary_path()).first, path);
+        auto new_path = format("{}../lib/kak:{}", split_path(get_kak_binary_path()).first, path);
         setenv("PATH", new_path.c_str(), 1);
     }
 }
diff --git a/libexec/kak/kak b/lib/kak/kak
similarity index 100%
rename from libexec/kak/kak
rename to lib/kak/kak
